src/parse.c

```diff
@@ -59,8 +59,20 @@
  * Quotes are escaped at configuration generation time as needed, as they might be part of passwords for example.
  * Escaping backslashes in the parser affects "netplan set" as it will always escape \'s from
  * the input and update YAMLs with all the \'s escaped again.
+ *
+ * g_strescape will also escape all the non-ascii chars in the range 0x7f-0xff. This affects non-ascii
+ * strings and will break utf-8 strings for example. So we add the entire range to the list of exceptions.
 */
-static char* STRESCAPE_EXCEPTIONS = "\"\\";
+static char* STRESCAPE_EXCEPTIONS =
+    "\"\\"
+    "\x7f\x80\x81\x82\x83\x84\x85\x86\x87\x88\x89\x8a\x8b\x8c\x8d\x8e\x8f\x90"
```

Comment:
todo: Can you please give some more evidence which characters this affects specifically?

E.g. according to https://www.charset.org/utf-8 the \x7f is a "Control character for DELETE", which is probably something that we actually want to have escaped, for security reasons. There might be more control characters that might need to be excluded.

Reply:
Ah yes, I added the 0x7f because it's mentioned in the non-ascii range here https://docs.gtk.org/glib/func.strescape.html but IT IS in the ascii table. So it will not be part of any multi-byte utf-8 code point I suppose.

The rest of the range affects all the multi-byte characters.